spec/factories/amendments.rb

```diff
@@ -0,0 +1,13 @@
+FactoryBot.define do
+  factory :amendment do
+    association :claim, factory: [:claim, :submitted]
+    association :created_by, factory: :dfe_signin_user
+    claim_changes { ActiveSupport::HashWithIndifferentAccess.new(teacher_reference_number: [generate(:teacher_reference_number).to_s, claim.teacher_reference_number]) }
```

Comment:
I'd like to ask for opinions on this. I've made it match the class returned by `ActiveModel::Dirty#previous_changes`, which is what we use to populate this attribute in the app. But I'm wondering

1.  whether it's the right thing to be storing in the database
2. whether the specs need to match the app code so exactly

Reply:
Spoke to Tekin about this and he said that the best thing would be to save a normal `Hash` instead of one with indifferent access, using the keys that come in `previous_changes`, which are strings. So I've made that change.

Aside: `serialize ..., Hash` actually [serializes using YAML](http://api.rubyonrails.org/classes/ActiveRecord/AttributeMethods/Serialization/ClassMethods.html#method-i-serialize), so that's how you‘d find out what the serialization looks like, not `to_s`.

Reply:
Aaah, righto, that makes sense